src/transforms/select.jl

```diff
@@ -2,6 +2,38 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+struct TableSelection{T} 
+  table::T
+  cols::Vector{Symbol}
+end
+
+Tables.istable(::Type{<:TableSelection}) = true
+Tables.columnaccess(::Type{<:TableSelection}) = true
+Tables.columns(ts::TableSelection) = ts
+Tables.columnnames(ts::TableSelection) = ts.cols
+Tables.getcolumn(ts::TableSelection, col::Int) =
+  Tables.getcolumn(ts.table, ts.cols[col])
+Tables.getcolumn(ts::TableSelection, col::Symbol) = 
+  Tables.getcolumn(ts.table, col)
+
+function Tables.schema(ts::TableSelection)
+  schema = Tables.schema(ts.table)
+  tbtypes = schema.types
+  tbnames = collect(schema.names)
+  selecttypes = tbtypes[indexin(ts.cols, tbnames)]
+  return Tables.Schema(ts.cols, selecttypes)
+end
+
+Tables.materializer(::Type{TableSelection{T}}) where {T} = 
+  Tables.materializer(T)
+
+Base.:(==)(a::TableSelection, b::TableSelection) = a.table == b.table
+
```

Comment:
```suggestion
```
We do not need this comparison operator defined, also it is not correct  given that selections are only equal if the columns selected are equal. Let's erase this definition here.

Reply:
Resolved in commit af43f28.